### PR TITLE
A64: Implement REV, REV32, and REV16

### PR DESCRIPTION
**Pull request by Thog**

### CODE REVIEW

File: src/frontend/A64/decoder/a64.inc [hunk 1]
```diff
@@ -286,11 +286,11 @@ INST(UnallocatedEncoding,    "",                                          "10111
 
 // Data Processing - Register - 1 source
 //INST(RBIT_int,               "RBIT",                                      "z101101011000000000000nnnnnddddd")
-//INST(REV16_int,              "REV16",                                     "z101101011000000000001nnnnnddddd")
-//INST(REV,                    "REV",                                       "z10110101100000000001-nnnnnddddd")
+INST(REV16_int,              "REV16",                                     "z101101011000000000001nnnnnddddd")
+INST(REV,                    "REV",                                       "z10110101100000000001-nnnnnddddd")
```
Comment (MerryMage) [inline src/frontend/A64/decoder/a64.inc]: `"z10110101100000000001onnnnnddddd"`

File: src/frontend/A64/translate/impl/data_processing_register.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * This software may be used and distributed according to the terms of the GNU
+ * General Public License version 2 or any later version.
+ */
+
+#include "frontend/A64/translate/impl/impl.h"
+
+namespace Dynarmic {
+namespace A64 {
+
+bool TranslatorVisitor::REV(bool sf, Reg Rn, Reg Rd)
```
Comment (MerryMage) [inline src/frontend/A64/translate/impl/data_processing_register.cpp]: `boof sf, bool opc_0, Reg Rn, Reg Rd`

then:

`if (!sf && opc_0) return UnallocatedEncoding();`

File: src/frontend/A64/translate/impl/data_processing_register.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    size_t datasize = sf ? 64 : 32;
+
+    IR::U32U64 operand = X(datasize, Rn);
+    IR::U32U64 result;
+    IR::U32 hihalf;
+    IR::U32 lohalf;
+
+
+    if (sf) {
+        IR::U32 lo = ir.LeastSignificantWord(operand);
```
Comment (MerryMage) [inline src/frontend/A64/translate/impl/data_processing_register.cpp]: Instead consider:

    IR::U64 hihalf = ir.And(ir.LogicalShiftRight(operand, ir.Imm8(8)), ir.Imm64(0x00FF00FF00FF00FF));
    IR::U64 lohalf = ir.And(ir.LogicalShiftLeft(operand, ir.Imm8(8)), ir.Imm64(0xFF00FF00FF00FF00));
    result = ir.Or(hihalf, lohalf);

**Comment by MerryMage** [comment]
Relevant test failure: https://travis-ci.org/MerryMage/dynarmic/jobs/331319924#L1226

**Comment by Thog** [comment]
The two issues should be fixed ^^

**Pull request merged**